### PR TITLE
ci: use reuseable workflows

### PR DESCRIPTION
**Pull request by huitseeker**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -67,49 +67,16 @@ jobs:
...
-        run: cargo build --target wasm32-unknown-unknown
-      - name: Check Rustfmt Code Style
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
-      - name: Check clippy warnings
-        run: cargo xclippy -D warnings
-      - name: Check benches compile under the flamegraph feature
-        run: cargo check --profile dev-ci --features "flamegraph" --benches
```
Comment (samuelburnham) [inline .github/workflows/rust.yml:98]: This flamegraph check isn't contained in the reusable workflow, so it would need a separate job

**Review by samuelburnham — CHANGES_REQUESTED** [review]

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -67,49 +69,16 @@ jobs:
       working-directory: ${{ github.workspace }}/lurk
```
Comment (samuelburnham) [inline .github/workflows/rust.yml:69]: We can also use the https://github.com/lurk-lab/ci-workflows/blob/main/.github/workflows/check-lurk-compiles.yml workflow here, or wait till it's more versatile as it only accepts a `path` and optional `package` keyword for the patch

**Review by samuelburnham — DISMISSED** [review]

**Comment by huitseeker** [comment]
Pausing work on this: we need to switch to https://github.com/dtolnay/rust-toolchain first, before we use our reusable workflows.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (huitseeker) [reply .github/workflows/rust.yml:69]: > wait till it's more versatile as it only accepts a path and optional package keyword for the patch

I'm not sure I understand what you're suggesting?

Comment (samuelburnham) [reply .github/workflows/rust.yml:68]: Future work could be to refine the `check-lurk-compiles` reusable workflow to account for things like arecibo having a `package` Cargo rename, different features, etc. It's not related to this PR

**Review by samuelburnham — APPROVED** [review]

**Pull request merged**